### ISSUE TITLE
RxError: query must be an object

### ISSUE THREAD

**Issue by gotjoshua** [pubkey/rxdb#2229]
## Case
Error Message when you pass a number to a string needs to be more explicit

https://codesandbox.io/s/vue-rxdb-w0txi?file=/src/components/home.vue

## Issue
Some of RxDBs error messages are delightfully explicit (like if you omit a required field)
But this one was very cryptic:
```
"RxError: query must be an object
Given parameters: {
queryObj:411}"
```
I eventually figured out that the data had numbers and i needed to cast the id (in order to comply with RxDB's restriction that all primaries must be of type string - why is that by the way?).

## Info
  - Environment: (Node.js/browser/electron/etc..)
  - Adapter: (IndexedDB/Localstorage/LevelDB/etc..)
  - Stack: (Typescript, Babel, Angular, React, etc..)

## Code
Full example is here:
https://codesandbox.io/s/vue-rxdb-w0txi?file=/src/components/home.vue

key part of the code is here:
```js
 const db = await RxDB.create({
      name: "main",
      adapter: "idb",
      multiInstance: true,
      queryChangeDetection: true,
      ignoreDuplicate: true
    });
    await db.collection({
      name: "items",
      schema: {
        version: 0,
        type: "object",
        properties: {
          _id: { type: "string", primary: true },
          name: { type: "string" },
          ID: { type: "number" },
          color: { type: "string" }
        },
        required: ["ID"]
      }
    });
    // this.$set(this, "db", db);
    await db.items.upsert({ _id: 411, name: "mike" }); // putting the 411 into quotes will fix the error
```
I would expect the error to indicate:
Type mismatch on insert:
_id is expected to be of type "string" 
instead is of type "number"

**Comment by pubkey** [comment]
Yes this is a good proposal. We should add this to the dev-mode checks.
